src/anyio/_core/_synchronization.py

```diff
@@ -46,11 +73,32 @@ class SemaphoreStatistics:
     tasks_waiting: int
 
 
+class Event:
```

Comment:
No. Static checkers are not interested in what `__new__` actually does, they only see that somebody is trying to instantiate this class.

Reply:
Mypy seems to have a problem with this, and I'm not sure how to resolve it:
```
src/anyio/_core/_synchronization.py:122: error: Cannot instantiate abstract class 'Event' with abstract attributes 'is_set', 'set', 'statistics' and 'wait'
```

Reply:
Ask the mypy people how they'd handle this situation?

Reply:
I think that the only reason why this didn't happen with `CapacityLimiter` too is because I don't instantiate the top level class anywhere in anyio code (tests are not type checked).

Reply:
Hum it is weird to read this class without method implementation @smurfix but if mypy disallows it.. I don't have a better solution

Reply:
Hum it is weird to read this class without method implementation @smurfix but if mypy disallows it.. I don't have a better solution

Reply:
I've made it so that it raises `NotImplementedError` just in case. I've also made the `CapacityLimiter` class non-abstract the same way.

Reply:
it is ok for me